portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java

```diff
@@ -71,8 +74,26 @@ public ServletPaths down(String path) {
 			return this;
 		}
 
-		return new ServletPaths(
-			_resourcePath.concat(normalizedPath), _servletContext);
+		int pos = normalizedPath.indexOf(CharPool.QUESTION);
+
+		if (pos != -1) {
+			normalizedPath = normalizedPath.substring(0, pos);
```

Comment:
I don't think this is right to simply throw away the query string.
For normalized path building, yes, we need to remove the query string. But shouldn't we add it back to the final normalized path?

Reply:
/cc @jonathanmccann @gregory-bretall, thoughts on @shuyangzhou's comment?

Reply:
@shuyangzhou: We tried leaving it there, and Windows complained about ? being an invalid character inside of a filename.

Reply:
@shuyangzhou: Or do you mean, add the query string to the return value `return new ServletPaths(downPathString, _servletContext)`?

Reply:
@shuyangzhou and @holatuwol I have made a version of the fix that adds the query string back in to the return value and it still works in my testing, does that seem like something we want?